### ISSUE TITLE
xamarin ios release package cannot launch

### ISSUE THREAD

**Issue by llzz19851985** [xamarin/xamarin-macios#17981]
My app can run well in debug,but when I dispatch with release package,it can install but cannot launch,I found the reason maybe is the code some where with AOT complier error,but i cannot found where is the code error,is there any log for this problem?thanks

**Comment by archergod** [comment]
I am facing similar error, and if I run my app in debug with Linker = Link Only SDK I can repeat the error. It seems linker is removing some function that are required to launch our app. Setting `Main Interface` in `info.plist`  though make it working

**Comment by archergod** [comment]
In my case it was some programming error from my side that I move `MainPage=new AppShell()` from constructor to `OnStart` event. But I notice that Visual Studio is moody in compiling the app and often mess with `info.plist` or some other setting and fail to either upload or make application start. 

The only solution that works for me is to Clean project, delete bin/obj manually. sometime restart Visual Studio and reconnect to Mac. Then it works fine.

You can also see Crash report from Apple Store, it has Open with Xcode (on Mac indeed). try that it sometime give information for debugging.

**Comment by llzz19851985** [comment]
@rolfbjarne what is the difference between the two method?now i just running the "Release" configuration locally

**Comment by llzz19851985** [comment]
@rolfbjarne yes,add --interpreter for mtouch it can work,if i disable the llvm,it also can work,But the performance is terrible!
I demoed llvm,--interpreter config performance,when no llvm and no --interpreter,run time spent 800ms,when add --interpreter The calculation time actually takes 150 seconds！when enable llvm without --interpreter,the time just 1ms,my test code as below:
private void Cal()
        {
            Stopwatch sw = new Stopwatch();
            sw.Start();
            int calCount = 20001;
            for (int i = 0; i < calCount; i++)
            {
                for (int j = 0; j < calCount; j++)
                {
                    double a = Math.PI * i / Math.Round(100.0);
                    double b = Math.PI * j / Math.Round(400.0);
                    double max = Math.Max(a, b);
                }
            }
            sw.Stop();
            TimeSpan ts2 = sw.Elapsed;
            Console.WriteLine($"=====================CalCount[{calCount}],Spent All Time {ts2.TotalMilliseconds:f2}ms.");
        }

**Comment by rolfbjarne** [comment]
OK, can you try this and see if it fixes the issue while keeping the performance acceptable:

```xml
<PropertyGroup>
    <MtouchInterpreter>-all</MtouchInterpreter>
</PropertyGroup>
```

**Comment by rolfbjarne** [comment]
@llzz19851985 please create a complete test project we can use to reproduce the performance issue, and we'll have a look.

**Comment by llzz19851985** [comment]
@rolfbjarne My computer is encrypted and cannot provide test sample，the test sample is very easy,just create any xamarin ios project and than use the code as follow:
private void Cal()
{
Stopwatch sw = new Stopwatch();
sw.Start();
int calCount = 20001;
for (int i = 0; i < calCount; i++)
{
for (int j = 0; j < calCount; j++)
{
double a = Math.PI * i / Math.Round(100.0);
double b = Math.PI * j / Math.Round(400.0);
double max = Math.Max(a, b);
}
}
sw.Stop();
TimeSpan ts2 = sw.Elapsed;
Console.WriteLine($"=====================CalCount[{calCount}],Spent All Time {ts2.TotalMilliseconds:f2}ms.");
}

**Comment by rolfbjarne** [comment]
I can't see any performance slowdown with the interpreter enabled like this:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net7.0-ios</TargetFramework>
    <OutputType>Exe</OutputType>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <SupportedOSPlatformVersion>13.0</SupportedOSPlatformVersion>
    <MtouchUseLlvm>true</MtouchUseLlvm>
    <MtouchInterpreter>-all</MtouchInterpreter>
    <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
  </PropertyGroup>
</Project>
```

vs:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net7.0-ios</TargetFramework>
    <OutputType>Exe</OutputType>
    <Nullable>enable</Nullable>
    <ImplicitUsings>true</ImplicitUsings>
    <SupportedOSPlatformVersion>13.0</SupportedOSPlatformVersion>
    <MtouchUseLlvm>true</MtouchUseLlvm>
    <!-- <MtouchInterpreter>-all</MtouchInterpreter> -->
    <RuntimeIdentifier>ios-arm64</RuntimeIdentifier>
  </PropertyGroup>
</Project>
```

If you see something else, we'll need a complete test project we can try, to make sure we're testing the same thing.

**Comment by llzz19851985** [comment]
@rolfbjarne the reason performance have no difference is you enable the **LLVM** !
In addition, this issue is about the problem that LLVM cannot launch and the reason.

**Comment by rolfbjarne** [comment]
> @rolfbjarne the reason performance have no difference is you enable the **LLVM** ! In addition, this issue is about the problem that LLVM cannot launch and the reason.

Are you saying that if you enable llvm and the interpreter like this:

```xml
<PropertyGroup>
    <MtouchUseLlvm>true</MtouchUseLlvm>
    <MtouchInterpreter>-all</MtouchInterpreter>
</PropertyGroup>
```

then your app does not launch?

**Comment by rolfbjarne** [comment]
> @rolfbjarne Please have a detailed look at my entire question and this message，thank you! [#17981 (comment)](https://github.com/xamarin/xamarin-macios/issues/17981#issuecomment-1504324851)

You can see the log from the device using the Console app like Apple explains here: https://support.apple.com/guide/console/log-messages-cnsl1012/mac#:~:text=If%20you%20don't%20see%20the%20log%20message%20details%2C%20click,in%20log%20messages%20and%20reports.

That might show the reason why your app doesn't launch.
